service/history/queues/scheduler.go

```diff
@@ -193,6 +193,19 @@ func (s *schedulerImpl) TaskChannelKeyFn() TaskChannelKeyFn {
 	return s.taskChannelKeyFn
 }
 
+// CommonSchedulerWrapper is an adapter that converts a common [task.Scheduler] to a [Scheduler] adding an empty
+// TaskChannelKeyFn implementation for schedulers that don't require this functionality.
+type CommonSchedulerWrapper struct {
+	tasks.Scheduler[Executable]
+}
+
+func (s *CommonSchedulerWrapper) TaskChannelKeyFn() TaskChannelKeyFn {
```

Comment:
Task rescheduler will call this, so that rescheduling for different channels won't block each other. so I think callback queue still needs this.

Reply:
oh I just realized upon failover, rescheduler need to immediately reschedule all tasks for the namespace that's failed over because the execution logic for that ns's tasks will change from standby to active. Waiting on the backoff interval (which maybe very long ~mins if the standby task has retried many time) for standby logic will block workflow from making progress for too long. https://github.com/temporalio/temporal/blob/main/service/history/queues/rescheduler.go#L164

We likely need to do some refactoring and make TaskChannelKey generic (but still be able to get Namespace from it). This can be done later when working on the replication part.

Reply:
I made the function injectable and injected a sane default.
We'll revisit this when we get to replication (I'll track).